Cubical/Categories/Constructions/TotalCategory.agda

```diff
@@ -0,0 +1,74 @@
+{-# OPTIONS --safe #-}
+module Cubical.Categories.Constructions.TotalCategory where
+
+open import Cubical.Foundations.Prelude
+open import Cubical.Foundations.HLevels
+
+open import Cubical.Data.Sigma
+
+open import Cubical.Categories.Category.Base
+open import Cubical.Categories.Functor
+open import Cubical.Categories.Displayed.Base
+open import Cubical.Categories.Displayed.Functor
+open import Cubical.Categories.Displayed.Instances.Terminal
+
+private
+  variable
+    ℓC ℓC' ℓD ℓD' ℓE ℓE' ℓCᴰ ℓCᴰ' ℓDᴰ ℓDᴰ' ℓEᴰ ℓEᴰ' : Level
+
+-- Total category of a displayed category
+module _ {C : Category ℓC ℓC'} (Cᴰ : Categoryᴰ C ℓCᴰ ℓCᴰ') where
+
+  open Category
+  open Categoryᴰ Cᴰ
+  private
+    module C = Category C
+
+  ∫C : Category (ℓ-max ℓC ℓCᴰ) (ℓ-max ℓC' ℓCᴰ')
+  ∫C .ob = Σ _ ob[_]
+  ∫C .Hom[_,_] (_ , xᴰ) (_ , yᴰ) = Σ _ Hom[_][ xᴰ , yᴰ ]
+  ∫C .id = _ , idᴰ
+  ∫C ._⋆_ (_ , fᴰ) (_ , gᴰ) = _ , fᴰ ⋆ᴰ gᴰ
+  ∫C .⋆IdL _ = ΣPathP (_ , ⋆IdLᴰ _)
+  ∫C .⋆IdR _ = ΣPathP (_ , ⋆IdRᴰ _)
+  ∫C .⋆Assoc _ _ _ = ΣPathP (_ , ⋆Assocᴰ _ _ _)
+  ∫C .isSetHom = isSetΣ C.isSetHom (λ _ → isSetHomᴰ)
+
+-- Total functor of a displayed functor
+module _ {C : Category ℓC ℓC'} {D : Category ℓD ℓD'}
+  {F : Functor C D} {Cᴰ : Categoryᴰ C ℓCᴰ ℓCᴰ'} {Dᴰ : Categoryᴰ D ℓDᴰ ℓDᴰ'}
+  (Fᴰ : Functorᴰ F Cᴰ Dᴰ)
+  where
+
+  open Functor
+  private
+    module Fᴰ = Functorᴰ Fᴰ
+
+  ∫F : Functor (∫C Cᴰ) (∫C Dᴰ)
+  ∫F .F-ob (x , xᴰ) = _ , Fᴰ.F-obᴰ xᴰ
+  ∫F .F-hom (_ , fᴰ) = _ , Fᴰ.F-homᴰ fᴰ
+  ∫F .F-id = ΣPathP (_ , Fᴰ.F-idᴰ)
+  ∫F .F-seq _ _ = ΣPathP (_ , (Fᴰ.F-seqᴰ _ _))
+
+-- Projections out of the total category
+module _ {C : Category ℓC ℓC'} {Cᴰ : Categoryᴰ C ℓCᴰ ℓCᴰ'} where
+  open Functor
+  open Functorᴰ
+
+  Fst : Functor (∫C Cᴰ) C
```

Comment:
Sounds good. We were trying to keep consistent with `Categories.Constructions.BinProduct`, which has the first projection functor out of the product category as `Fst` as well. 
We should probably then change the instance in `BinProduct` too then

Reply:
Although I don't immediately see any good candidate for `SOMETHING`. Maybe something like `fstProjection` or `π₁F`?

Reply:
I think we should either leave it as `Fst` to match BinProduct or change both of them to `fst` to match Data.Sigma and then use them as qualifies imports when you need to disambiguate

Reply:
I see. I think the solution with qualified imports is better, guessing that it is very unlikely that you want to have both available, ```fst``` from ```Data.Sigma``` and this one. For example in this file, it makes a lot of sense to me to qualify ```Data.Sigma```.  Feel free to disagree and go with some other name, my main problem with ```Fst``` is that it starts in upper case for disambiguation (maybe we should fix a convention for disambiguation which is already in use, like appending apostrophies to a name). Another option might be to use ```π₁```.

Reply:
Qualified imports could work here, but they cause more problems than the seem to solve. Choosing the name `fst` for both here and `BinProduct` requires renamings in both `Data.Sigma` and `Prelude` because Sigma types are defined as primitives. 

That is, anytime we simultaneously import any of `Prelude`, `Data.Sigma`, `Constructions.TotalCategory`, or `Constructions.BinProduct` we need to add several qualified imports. This is every import site for `BinProduct`. These qualified imports unnecessarily complicates imports in files that are unrelated to this PR. It seems like `Fst` is preferable to make this PR a more modular addition to the repo. 

I think it makes the most sense to keep the naming of this functor aligned with the projection in `BinProduct`. So as an alternative, we can go with `π₁` in each of these spots and correspondingly adapt any code that imports `BinProduct`

Reply:
I don't think what you're describing is actually a problem, Steven, it's normal to have to use a lot of qualified imports IMO.

But I agree that this should really be combined with fixing BinProduct as well in a separate PR. So I recommend not changing the names for now and then making an issue where we can bikeshed this more.

Reply:
Yes - agreed :-)
I'll merge it.